### PR TITLE
homework.lambda.part3.exercise2 (kornaushenkov)

### PR DESCRIPTION
**Pull request by EugeneGrim**

### CODE REVIEW

File: src/test/java/lambda/part3/exercise/Exercise2.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        /**
+         * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
+         * Для добавления в новый список каждый элемент преобразовывается с использованием заданной функции.
+         * ([T], (T -> R)) -> [R]
+         * @param mapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> map(Function<T, R> mapping) {
+            return new MapHelper<>(source.stream()
+                    .map(mapping)
+                    .collect(Collectors.toList()));
```
Comment (vkovrovEPAM) [inline src/test/java/lambda/part3/exercise/Exercise2.java]: It is not a task on Streams, use forEach

File: src/test/java/lambda/part3/exercise/Exercise2.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+         * @param flatMapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> flatMap(Function<T, List<R>> flatMapping) {
+            return new MapHelper<>(source.stream()
+                    .map(flatMapping)
+                    .reduce((r1, r2) -> {
+                        r1.addAll(r2);
+                        return r1;
+                        })
+                    .orElseGet(null));
```
Comment (vkovrovEPAM) [inline src/test/java/lambda/part3/exercise/Exercise2.java]: It is not a task on Streams, use forEach

**Review by vkovrovEPAM — CHANGES_REQUESTED** [review]

File: src/test/java/lambda/part3/exercise/Exercise2.java (hunk 1 above)
Comment (EugeneGrim) [reply src/test/java/lambda/part3/exercise/Exercise2.java]: Fixed

File: src/test/java/lambda/part3/exercise/Exercise2.java (hunk 2 above)
Comment (EugeneGrim) [reply src/test/java/lambda/part3/exercise/Exercise2.java]: Fixed

**Review by vkovrovEPAM — APPROVED** [review]

**Review by Kostyanoid — APPROVED** [review]

**Pull request merged**